presto-main/src/test/java/com/facebook/presto/operator/TestPartitionedOutputOperator.java

```diff
@@ -0,0 +1,340 @@
+                buffers,
+                dataSize,
+                () -> new SimpleLocalMemoryContext(newSimpleAggregatedMemoryContext(), "test"),
+                SCHEDULER);
+    }
+
+    private Page buildPage(
+            List<Type> types,
+            int positionCount,
+            boolean addPreComputedHashBlockAtBegin,
+            boolean addNullBlockAtEnd,
+            boolean allowNulls,
+            boolean useBlockView,
+            Optional<String> wrappings)
+    {
+        int channelCount = types.size();
+        int preComputedChannelCount = (addPreComputedHashBlockAtBegin ? 1 : 0);
+        int nullChannelCount = (addNullBlockAtEnd ? 1 : 0);
+
+        Block[] blocks = new Block[channelCount + preComputedChannelCount + nullChannelCount];
+
+        if (addPreComputedHashBlockAtBegin) {
+            blocks[0] = buildBigintBlock(positionCount, false);
+        }
+
+        for (int i = 0; i < channelCount; i++) {
+            blocks[i + preComputedChannelCount] = buildBlockWithType(types.get(i), positionCount, allowNulls, useBlockView, wrappings);
+        }
+
+        if (addNullBlockAtEnd) {
+            blocks[channelCount + preComputedChannelCount] = buildNullBlock(BIGINT, positionCount);
+        }
+
+        return new Page(positionCount, blocks);
+    }
+
+    Block buildBigintBlock(int positionCount, boolean allowNulls)
+    {
+        return createLongsBlock(
+                IntStream.range(0, positionCount)
+                        .mapToObj(i -> allowNulls && i % 7 == 0 ? null : RANDOM.nextLong())
+                        .collect(Collectors.toList()));
+    }
+
+    Block buildBlockWithType(Type type, int positionCount, boolean allowNulls, boolean isView, Optional<String> wrappings)
+    {
+        Block block = null;
+
+        if (isView) {
+            positionCount *= 2;
+        }
+
+        if (type == BIGINT || type instanceof DecimalType && ((DecimalType) type).isShort()) {
+            block = buildBigintBlock(positionCount, allowNulls);
+        }
+
+        if (isView) {
+            positionCount /= 2;
+            int offset = positionCount / 2;
+            block = block.getRegion(offset, positionCount);
+        }
+
+        if (wrappings.isPresent()) {
+            block = buildDictRleBlock(block, positionCount, wrappings);
+        }
+
+        return block;
+    }
+
+    DictionaryBlock buildDictionaryBlock(Block dictionary, int positionCount)
+    {
+        int[] ids = IntStream.range(0, positionCount).map(i -> RANDOM.nextInt(dictionary.getPositionCount() / 10)).toArray();
+        return new DictionaryBlock(dictionary, ids);
+    }
+
+    RunLengthEncodedBlock buildRleBlock(Block block, int positionCount)
+    {
+        checkArgument(block.getPositionCount() > 0);
+        return new RunLengthEncodedBlock(block.getRegion(block.getPositionCount() / 2, 1), positionCount);
+    }
+
+    Block buildDictRleBlock(Block block, int positionCount, Optional<String> wrappings)
```

Comment:
We also don't need to use this method here. buildDictRleBlock was created to create multiple level nested Dictionary/Rle blocks to test block flattening and it's not our main goal here. We only need to keep the buildDictionaryBlock() and buildRleBlock().

Reply:
done